### DISCUSSION TITLE
Potential problem with ‘Act of Communication by Media’ definition

### DISCUSSION THREAD

**Discussion by gregfowlerphd** [CommonCoreOntology/CommonCoreOntologies#335] [Announcements]
The definition says that an ACBM is an AC ‘in which some Artifact is used to transfer an Information Bearing Entity from sender(s) to receiver(s)’. I admit I might be confused here, but isn’t it an ICE rather than an IBE that is transferred in case of, say, email messaging (which is one of the subclasses of ACBM)?

**Comment by swartik** [comment]
In support of @gregfowlerphd's point, superclass Act of Communication's definition says an ICE is transferred.

In opposition to @gregfowlerphd's point, can a Material Artifact operate on mere information? Doesn't a physical object require physical objects as inputs and produce physical objects as outputs? Then again, the physical object that is the input to a radio broadcast is waves of air pressing against a microphone. That's not much of a physical object.

So basically, I can see arguments on both sides, and I need someone brighter than me to tell me how to use CCO.

**Comment by gregfowlerphd** [comment]
Regarding @swartik's second point, I would think that a material artifact can operate on mere information in the same way as any other material entity--i.e., _by_ operating on physical objects. When I want to transfer information from my mind to yours, for instance, I must operate on physical objects to do so (as I do when I transfer such information by speaking).

**Comment by swartik** [comment]
In subclasses Email Messaging, Facsimile Transmission, Instant Messaging, Mailing, Text Messaging, and Webcast, the information content is explicitly represented in an IBE prior to being transferred. In Radio Broadcast, Telephone Call, and Television Broadcast it is not, at least for live broadcasts. Is this dichotomy a contradiction, or is it irrelevant?

Personally I prefer to model using ICEs, but I've encountered persons who disagree with me, saying inputs and output of a process must be physical. I will offer this: transferring an IBE is something you do to a book or a letter. An ACBM does not transfer the physical representation of an email message. It creates a copy of the original. Both the original and the copy are carriers of the same ICE. If the definition of ACBM is to continue using the word "transfer", it should use ICE, not IBE.

If a process must have a physical effect, the effect of an ACBM is then either:
- The creation of an IBE (sending email), or
- The change of location of an IBE (mailing a letter).

**Comment by cameronmore** [comment]
This touches on my concerns in #293 and @alanruttenberg 's in #94  , what does it mean to send and receive information? I think it means to send and receive some bearer of the information, whether sound waves or electrical pulses.

I think it's entirely reasonable for a process to have an ICE as an output, but since ICEs can be found via their bearers, it follows that processes which produce ICEs also produce bearers, even if those bearers are not enduring (like paper which is shredded at the end of a meeting).

How does an artifact operate on information? Well, take, for instance, a volume knob on a radio. It is a cylinder that spins from 0 to 10, and bears some ICE that tells us what level of volume the radio is able to produce for us listening to it. When we adjust the knob, we are changing some mechanical process. Here's the way one type of knob works:

> The volume knob works as a variable resistor. It increases and decreases resistance from electrical signals when you turn it in a direction. When you turn the volume down, you increase the resistance allowing fewer volts to get to the speaker. Reducing the volts makes the speaker quieter. ([source](https://www.wellpcb.com/Volume-Control-Circuit-for-Speaker-Step-by-Step-Guide-to-Enjoy-Your-Music.html))

So, while it seems that an artifact is operating on some input information, it's really operating on some way the bearer bears that information (the level to which the knob is set, which correlates which some amount of electrical current).

I like modeling with ICEs, and the bearers become relevant when ICEs are transferred across information systems, through or between artifacts, between computers over a network, and so on. And the real life, fine grained story of what happens to those IBEs can be very complicated, so often we just abstract away.

Regardless, the definitions in this part of the hierarchy need work, and I think the act of communication by media or by artifact should be made equivalence classes, given the concern that it seems one cannot communicate through media at the same time as apologizing or denying (unless you double type an instance). Act of communication by artifact is always related via parthood to the other communication classes.

**Comment by gregfowlerphd** [comment]
@cameronmore: Having read and thought for awhile about your comment, I'm still concerned about the definition for ACBM. In your comment, you seem to suggest that in every case of sending and receiving information, and hence in every ACBM, there's an IBE that's both sent by the sender and received by the receiver, in accordance with the definition for ACBM. But I'm not sure that's accurate.

Consider a case of email messaging. On my understanding of how email works (which I fully admit is rough, somewhat simplistic, and possibly mistaken!), in such a case: A signal, which is a bearer of the relevant ICE, is sent to and received by an email server. The ICE is then stored, however briefly, as an electrical state of the server. Finally, the server sends a signal--which, like the first signal, is a bearer of the relevant ICE--to the recipient of the email.

If that's right, then in a case of email messaging, the definition for ACBM seems to get things wrong, since there's no IBE that's both sent by the sender and received by the receiver. Instead, there are (at least) three IBEs: the two signals and the server's electrical state.

So I'm inclined to think that the ACBM definition should refer to ICEs rather than IBEs. What do you think?

**Comment by cameronmore** [comment]
@gregfowlerphd no disagreements here, I suppose I mean that there's always a bearer, but there may be many bearers, some which copy the pattern of bits from one to the next, transforming the 'data' from one format to another, ultimately to be displayed on a screen to someone else. 

In the cyber domain particularly, you can have as many or as little bearers as you need depending on the level of granularity you take, which lurks in the background here.

**Comment by gregfowlerphd** [comment]
@cameronmore: Thanks for your reply, Cameron. Your point about levels of granularity in the cyber domain is interesting. I think it might provide another reason to favor ICEs rather than IBEs here, lest email messaging count as an ACBM at some levels of granularity but not at others.

**Comment by cameronmore** [comment]
Yeah, would love to get clearer on the sending and receiving over on the other issue

**Comment by alanruttenberg** [comment]
This is not an issue of granularity. @gregfowlerphd's understanding is correct. When you say you are sending an IBE, a material entity, then you are sending some physical thing. You send an IBE when you send a snail-mail letter in an envelope or when you ship a flash drive to someone. There is no singular IBE for sending an email message. 

When analyzing this we need to focus on what the subject of our assertions are. What matters when sending an email is that there is a copy created on the other end. A copy is a concretization of an ICE. We are not interested in any particular concretization or bearer, since no matter what they are (and in modern computer systems there will be many between sender and receiver), successful sending of an email message means the content has been successfully sent.

I understand that it may be confusing to work with ICEs - they seem invisible in some sense. But this is the nature of any generically dependent continuant. A GDC lives by it's concretizations and they live by their bearers. Nonetheless we represent the GDCs when we mean to make assertions that are about what is the same across all concretizations and bearers, and that is the content, in the case of ICEs.

**Comment by alanruttenberg** [comment]
In the case where you send a physical letter you care about the IBE because you have to put a stamp on it, and because it may be the only bearer of a concretization of the content. But the IBE is conveying (by carrying a copy) an ICE. When you tell your friend about what the letter said, it is the ICE you are talking about. These days, for the most part, the relating of the content will not differ between whether you've received a physical letter or whether you received the letter over email. The only thing might differ is that it is notable these days that content is sent, between friends, by snail-mail. So you might remark on that in talking about the message. This is a case of the [medium being (part of) the message](https://en.wikipedia.org/wiki/The_medium_is_the_message).

**Comment by alanruttenberg** [comment]
Another angle on this is counting. You write an email and there are ten addressees on it. Let's suppose everybody reads it. Are they reading ten different messages or one? The sender would say that they sent the *same (one) email* to the ten recipients, and the recipients would say they all received the same one email. If you think there are ten email messages, how does that play out? In subsequent communications the initial recipients typically talk about the content of the email.

Only occasionally may they mention some aspect relating to a bearer, such as mentioning they read it on their phone and so couldn't see the whole of an attached presentation. And in that case they aren't talking about a different message, they are talking about a particular process of reading, concretization and bearer, a concretization and bearer the author of the email certainly didn't have specifically in mind when authoring the message.

**Comment by gregfowlerphd** [comment]
@alanruttenberg: Really appreciate you weighing in. And sorry to step on your toes! The problem I raised here does seem to be the same as that discussed in your #94.

@cameronmore: Having read your #293, I agree that it would be good to get clearer on sending and receiving. While I think that issue is orthogonal to the issue of whether an IBE or an ICE is transferred in an ACBM, it might also affect the ACBM definition, given that the definition mentions sender(s) and receiver(s).

**Comment by swartik** [comment]
Regarding @alanruttenberg's point about counting, under some definitions of "email message" recipients do not receive the same ICE. If "email message" includes header content, what is received includes some routing and timing information, which differs from recipient to recipient. CCO's definition source for `Email Message` is Wikipedia, which cites RFC 5322, which considers headers part of an email message.

I realize we're really discussing the intuitive notion of "email message": the few header lines you see in your inbox, plus the body. I just want to point out that the ICE for an email message is quite complex, and changes during the time between sending and receiving. Here are two things to ponder:

1. If the sender includes Bcc: recipients, they aren't part of a receiver's ICE.
2. My organization has strict policies on email attachments. I may not receive the same message you do.

In practice I doubt a single "email message" ICE will generically depend on multiple independent continuants or be concretized into multiple specifically dependent continuants.

**Comment by gregfowlerphd** [comment]
@swartik: Very interesting thoughts!

You say that 'under some definitions of "email message" recipients do not receive the same ICE'. I assume that in saying 'the same ICE', you're considering only the total information content of the message(s)? Because presumably there will be some ICEs both sent and received.* For example, if I compose an email containing the sentence 'ICEs are GDCs', enter your email address into the 'To' box, and hit send, then it seems that the information content of that sentence is something I send and you receive. (Moreover, that content will, I think, generically depend on both an independent continuant on my end and one on yours, as well as being concretized by both an SDC on my end and one on yours.)

As far as the intuitive notion of an email message goes, one might attempt to identify such a message with the total information content shared by sender and receiver(s). That is, one might attempt to identify it with the (aggregate) ICE that consists of all and only those ICEs that are both sent and received.

(The cases you suggest pondering constitute potential objections to this identification. However, I'm not sure these objections are successful. On the intuitive notion of an email message, I think, an attachment is not part of the message; instead, it is something _attached to_ the message. Similarly, one might hold that on the intuitive notion, only the content of the main body (or, perhaps, the main body plus the subject) is part of an email message, and thus that recipients' email addresses, including those of Bcc: recipients, are not.)

Interestingly, CCO does not go this route. Instead, in CCO, an email message is taken to be, somewhat oddly in my view, a material entity of a particular sort. (See the Email Message class and its subclass relations.)

In any case, as interesting as they are, I think the issues here are orthogonal to the initial point (though they are certainly relevant to the particular way @alanruttenberg framed his counting point), which concerned the definition of 'Act of Communication by Media':

> An Act of Communication in which some Artifact is used to transfer an Information Bearing Entity from sender(s) to receiver(s).

The point was that what appears to be _transferred_ in certain ACBMs is an ICE rather than an IBE. Email messaging (a subclass of ACBM) was brought up as a case in point: In a case of email messaging, there is no IBE transferred from sender(s) to receiver(s), but there is an ICE so transferred.** This seems to be so regardless of what precisely we take an email message to be or to contain.


*Probably worth noting that we're both here using 'sends' and 'receives' differently from CCO. In CCO, the range of the object properties with those labels is Act of Communication, not ICE. I'll continue to do so in the remainder, but worth flagging.

**In fact, there might be many such ICEs, corresponding, for instance, to the information content of each sentence in the email. But that's neither here nor there.

**Comment by swartik** [comment]
@gregfowlerphd I do agree with you that we should focus this discussion on the original point. I hope to continue this discussion, but I think we should do it in another issue addressing what we mean by "email message".

So back to the original point: Is there anyone who disagrees that an Act of Communication By Media transfers an ICE rather than an IBE? How about the following as a revised definition:

> An Act of Communication in which Artifacts are used to transfer an Information Content Entity from sender(s) to receiver(s).

along with an elucidation:

> An ACBM transfers an Information Content Entity that generically depends on an Information Bearing Entity created by sender(s). Assuming the transfer does not corrupt the communication, receiver(s) receive the same Information Content Entity,  although it may generically depend on a different Information Bearing Entity.

This pushes away all the discussion about packaging an email message and RFC 5322. As I said, that's another issue.

**Comment by gregfowlerphd** [comment]
@swartik: Everyone who's been involved in this discussion seems to agree that an ACBM transfers an ICE rather than an IBE. I'm also pretty sure they'd all see your revised definition as an improvement. However, @cameronmore (if I understand him correctly) is hesitant about such a revision being made at this point because he believes the issues he's raised concerning the **sends** and **receives** object properties (see his #293) might also impact the correct definition for ACBM.

I do think the discussion of what an email message is should continue and that your proposal to create another issue for that topic is a good one.

**Comment by cameronmore** [comment]
Right, if ICEs can only be 'transferred' by their material bearers, then 'transfers an ICE' is a shortening of 'transferred an ICE by some process of spatial movement on behalf of the bearers of that ICE, which may be one or multiple,' and I just want to make sure that the decisions made here make sense given the concerns I have with those properties.

**Comment by alanruttenberg** [comment]
@cameronmore there is no guaranteed spatial movement of material bearers, and in most cases there is none. What we call "transfer" in the case of an ICE is a series of concretizations/copies on successively different usually stationary bearers. You don't have to move anything to have an image on one screen pop up on another. Where there is movement is in the propagation of electromagnetic signals (but see below re: dominoes, which is a better analogy for wave propagation). But we don't consider electromagnetic waves to be material entities. Now if you want to move deeper and get closer to movement you would need to get into how electromagnetic signals propagate through materials, since at least some of the signals will be through material. But even then, it isn't the case that an individual electron starts at one end of the wire and leaves the other end. So yes, there is some wiggling of electron and other atoms' positions, but one would be hard pressed to argue *that* is the relevant movement of material bearers 

I think even using the language of movement is suspect. We're used to the idea of transfer meaning movement but copying is not moving. The appearance of movement here is similar to the perceived motion in a video which your brain synthesizes from a series of still images.

Or to give another analogy, suppose we cross a stream by hopping from rock to rock. No rock is moving us in the same way a package is moved by a truck carrying while it while it travels.

Or for yet another analogy think of a cascade of dominoes tipping each other in a row. It *looks like* something is moving, in that the tipping happens in successive locations, but none of the dominoes moved from beginning to end.

I'll only briefly comment on the RFC business. You have to ask yourself is there something at the receiver that is identical to what was at the sender, and the answer is yes. In a textual message, the wording of the body is identical. That is closest to the ICE we are talking about when we send an email. There may be other ICEs that get created that have that as part, but they don't change what the sender intended the recipient to receive. That part is the content.

**Comment by cameronmore** [comment]
@alanruttenberg completely right re the movement of bearers, it gets very complex very quickly.

**Comment by neilotte** [comment]
@alanruttenberg @cameronmore @gregfowlerphd I'm converting this thread,  #94 , and #293 into discussion threads so that these matters are tracked as part of the information discussions led by Carter Benson. Thanks.
